src/main/java/org/threeten/extra/chrono/InternationalFixedDate.java

```diff
@@ -0,0 +1,624 @@
+     * Instead of returning an invalid result, the last valid day of the month, 2006-13-07, is selected instead.
+     * <p>
+     * This instance is immutable and unaffected by this method call.
+     *
+     * @param monthsToAdd the months to add, may be negative
+     * @return a {@code InternationalFixedDate} based on this date with the months added, not null
+     * @throws DateTimeException if the result exceeds the supported date range
+     */
+    @Override
+    InternationalFixedDate plusMonths (final long monthsToAdd) {
+        if (monthsToAdd == 0) {
+            return this;
+        }
+
+        long calcMonths = Math.addExact (getProlepticMonth (), monthsToAdd);
+        int newYear = ChronoField.YEAR.checkValidIntValue (Math.floorDiv (calcMonths, InternationalFixedChronology.MONTHS_IN_YEAR));
+        int newMonth = Math.toIntExact (calcMonths % InternationalFixedChronology.MONTHS_IN_YEAR) + 1;
+
+        return resolvePreviousValid (newYear, newMonth, getDayOfMonth ());
+    }
+
+    @Override
+    public InternationalFixedDate minus (final TemporalAmount amount) {
+        return (InternationalFixedDate) amount.subtractFrom (this);
+    }
+
+    @Override
+    public InternationalFixedDate minus (final long amountToSubtract, final TemporalUnit unit) {
+        return (amountToSubtract == Long.MIN_VALUE ? plus (Long.MAX_VALUE, unit).plus (1, unit) : plus (-amountToSubtract, unit));
+    }
+
+    //-------------------------------------------------------------------------
+    @Override  // for covariant return type
+    @SuppressWarnings ("unchecked")
+    public ChronoLocalDateTime<InternationalFixedDate> atTime (final LocalTime localTime) {
+        return (ChronoLocalDateTime<InternationalFixedDate>) ChronoLocalDate.super.atTime (localTime);
+    }
+
+    @Override
+    public long until (final Temporal endExclusive, final TemporalUnit unit) {
+        return until (InternationalFixedDate.from (endExclusive), unit);
+    }
+
+    @Override
+    long until (final AbstractDate end, final TemporalUnit unit) {
+        if (unit instanceof ChronoUnit) {
+            InternationalFixedDate ifxEnd = InternationalFixedDate.from (end);
+            switch ((ChronoUnit) unit) {
+                case YEARS:
+                    return yearsUntil (ifxEnd);
+                case DECADES:
+                    return yearsUntil (ifxEnd) / YEARS_IN_DECADE;
+                case CENTURIES:
+                    return yearsUntil (ifxEnd) / YEARS_IN_CENTURY;
+                case MILLENNIA:
+                    return yearsUntil (ifxEnd) / YEARS_IN_MILLENNIUM;
+                default:
+                    break;
+            }
+        }
+
+        return super.until (end, unit);
+    }
+
+    /**
+     * Get the number of years from this date to the given day.
+     *
+     * @param end The end date.
+     * @return The number of years from this date to the given day.
+     */
+    long yearsUntil (final InternationalFixedDate end) {
```

Comment:
Unfortunately, the calculation is more complicated than this - you _aren't_ "simulating the effect of inserting the leap day".